python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,252 @@
+                client, model_info = self._create_client(azure_ai_inference_settings)
+            except ValidationError as e:
+                raise ServiceInitializationError(f"Failed to validate Azure AI Inference settings: {e}") from e
+
+        super().__init__(
+            ai_model_id=model_info.model_name,
+            service_id=service_id or model_info.model_name,
+            client=client,
+        )
+
+    async def get_chat_message_contents(
+        self,
+        chat_history: ChatHistory,
+        settings: AzureAIInferenceChatPromptExecutionSettings,
+        **kwargs: Any,
+    ) -> list[ChatMessageContent]:
+        """Get chat message contents from the Azure AI Inference service.
+
+        Args:
+            chat_history: A list of chats in a chat_history object.
+            settings: Settings for the request.
+            kwargs: Optional arguments.
+
+        Returns:
+            A list of chat message contents.
+        """
+        response: ChatCompletions = await self.client.complete(
+            messages=self._format_chat_history(chat_history),
+            model_extras=settings.extra_parameters,
+            **settings.prepare_settings_dict(),
+        )
+        response_metadata = self._get_metadata_from_response(response)
+
+        return [self._create_chat_message_content(response, choice, response_metadata) for choice in response.choices]
+
+    async def get_streaming_chat_message_contents(
+        self,
+        chat_history: ChatHistory,
+        settings: AzureAIInferenceChatPromptExecutionSettings,
+        **kwargs: Any,
+    ) -> AsyncGenerator[list[StreamingChatMessageContent], Any]:
+        """Get streaming chat message contents from the Azure AI Inference service.
+
+        Args:
+            chat_history: A list of chats in a chat_history object.
+            settings: Settings for the request.
+            kwargs: Optional arguments.
+
+        Returns:
+            A list of chat message contents.
+        """
+        response: AsyncStreamingChatCompletions = await self.client.complete(
+            stream=True,
+            messages=self._format_chat_history(chat_history),
+            model_extras=settings.extra_parameters,
+            **settings.prepare_settings_dict(),
+        )
+
+        async for chunk in response:
+            if len(chunk.choices) == 0:
+                continue
+            chunk_metadata = self._get_metadata_from_response(chunk)
+            yield [
+                self._create_streaming_chat_message_content(chunk, choice, chunk_metadata) for choice in chunk.choices
+            ]
+
+    def _create_client(
+        self, azure_ai_inference_settings: AzureAIInferenceSettings
+    ) -> tuple[ChatCompletionsClientAsync, ModelInfo]:
+        """Create the Azure AI Inference client.
+
+        Client is created synchronously to check the model type before creating the async client.
+        """
+        chat_completions_client_sync = load_client_sync(
```

Comment:
TLDR: We need to check if the endpoint provided is actually pointing to the model of the right type. We want to detect the potential case where the user passes an embedding model endpoint to the chat service.

Details:
The inference SDK provides two `load_client` methods. The async version returns an async client, while the sync version returns a sync client. And the `load_client` method takes the endpoint and internally decides the right client to return (i.e. Chat v.s. Embedding). 

We cannot use the async version of the `load_client` method since we are instantiating the client in the constructor. So, we must use the sync `load_client` method to determine the model type of the endpoint.